receiver/awslambdareceiver/handler.go

```diff
@@ -115,13 +161,142 @@ func (*s3Handler[T]) parseEvent(raw json.RawMessage) (event events.S3EventRecord
+	return cwEvent
+}
+
+func (c *cwLogsSubscriptionHandler) handle(ctx context.Context, event json.RawMessage) error {
+	var log events.CloudwatchLogsEvent
+	if err := gojson.Unmarshal(event, &log); err != nil {
+		return fmt.Errorf("failed to unmarshal cloudwatch event log: %w", err)
+	}
+
+	decoded, err := base64.StdEncoding.DecodeString(log.AWSLogs.Data)
+	if err != nil {
+		return fmt.Errorf("failed to decode data from cloudwatch logs event: %w", err)
+	}
+
+	var reader *gzip.Reader
+	reader, err = gzip.NewReader(bytes.NewReader(decoded))
+	if err != nil {
+		return fmt.Errorf("failed to decompress data from cloudwatch subscription event: %w", err)
+	}
+
+	defer reader.Close()
+
+	var decodedData bytes.Buffer
+	_, err = decodedData.ReadFrom(reader)
+	if err != nil {
+		return fmt.Errorf("failed to read decompressed data from cloudwatch subscription event: %w", err)
+	}
+
+	data, err := c.unmarshal(decodedData.Bytes())
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal CloudWatch logs: %w", err)
+	}
+	if err := c.consumer(ctx, data); err != nil {
+		return checkConsumerErrorAndWrap(err)
+	}
+
+	return nil
+}
+
+// cwLogsToPlogs implements unmarshalFunc for plog.Logs.
+// This defines the built-in behavior for CloudWatch subscription filter events when no encoding extension is provided.
+func cwLogsToPlogs(data []byte) (plog.Logs, error) {
+	var cwLog events.CloudwatchLogsData
+	err := gojson.Unmarshal(data, &cwLog)
+	if err != nil {
+		return plog.Logs{}, fmt.Errorf("failed to unmarshal data from cloudwatch logs event: %w", err)
+	}
+
+	logs := plog.NewLogs()
+	rl := logs.ResourceLogs().AppendEmpty()
+	resourceAttrs := rl.Resource().Attributes()
+	resourceAttrs.PutStr(string(conventions.CloudProviderKey), conventions.CloudProviderAWS.Value.AsString())
+	resourceAttrs.PutStr(string(conventions.CloudAccountIDKey), cwLog.Owner)
+	resourceAttrs.PutStr(string(conventions.AWSLogGroupNamesKey), cwLog.LogGroup)
+	resourceAttrs.PutStr(string(conventions.AWSLogStreamNamesKey), cwLog.LogStream)
+
+	sl := rl.ScopeLogs().AppendEmpty()
+	sl.Scope().SetName(metadata.ScopeName)
+
+	for _, event := range cwLog.LogEvents {
+		logRecord := sl.LogRecords().AppendEmpty()
+		// pcommon.Timestamp is a time specified as UNIX Epoch time in nanoseconds
+		// but timestamp in cloudwatch logs are in milliseconds.
+		logRecord.SetTimestamp(pcommon.Timestamp(event.Timestamp * int64(time.Millisecond)))
+		logRecord.Body().SetStr(event.Message)
+	}
+
+	return logs, err
+}
+
+// bytesToPlogs implements unmarshalFunc for plog.Logs.
+// This defines the built-in behavior for S3 events when no encoding extension is provided.
+func bytesToPlogs(data []byte) (plog.Logs, error) {
+	logs := plog.NewLogs()
+	rl := logs.ResourceLogs().AppendEmpty()
+	sl := rl.ScopeLogs().AppendEmpty()
+	sl.Scope().SetName(metadata.ScopeName)
+
+	lr := sl.LogRecords().AppendEmpty()
+	lr.Attributes().PutInt("content.length", int64(len(data)))
```

Comment:
```suggestion
```

Doesn't seem to add any value - can't we easily determine the length from the body?

Reply:
I originally thought it might help with some efforts like streaming as S3 bound files can be large, but given that we add everything to plog, yes it doesn't add value. So will remove it.